From: "Uladzislau Rezki (Sony)" <urezki@gmail.com>
Subject: mm: vmalloc: Fix a warning in the crash_save_vmcoreinfo_init()
Date: Thu, 11 Jan 2024 20:23:29 +0100

The vmcoreinfo_append_str() function expects "long unsigned int" type as a
second argument(0x%lx) to print a beginning of vmalloc start address which
is defined as a VMALLOC_START macro.

For some architectures it can be considered as "int" type, for example m68
generates a compile warning message.  To fix it cast a second argument to
"unsigned long".

Link: https://lkml.kernel.org/r/20240111192329.449189-1-urezki@gmail.com
Fixes: 9bdb180b2d ("mm/vmalloc: remove vmap_area_list")
Signed-off-by: Uladzislau Rezki (Sony) <urezki@gmail.com>
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202401120218.y469Puyf-lkp@intel.com/
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/crash_core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/kernel/crash_core.c~mm-vmalloc-remove-vmap_area_list-fix
+++ a/kernel/crash_core.c
@@ -748,7 +748,7 @@ static int __init crash_save_vmcoreinfo_
 	VMCOREINFO_SYMBOL_ARRAY(swapper_pg_dir);
 #endif
 	VMCOREINFO_SYMBOL(_stext);
-	vmcoreinfo_append_str("NUMBER(VMALLOC_START)=0x%lx\n", VMALLOC_START);
+	vmcoreinfo_append_str("NUMBER(VMALLOC_START)=0x%lx\n", (unsigned long) VMALLOC_START);
 
 #ifndef CONFIG_NUMA
 	VMCOREINFO_SYMBOL(mem_map);
_
